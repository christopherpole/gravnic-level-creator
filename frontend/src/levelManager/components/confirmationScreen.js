import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Button from 'common/button';
import { cancelConfirmation, confirmConfirmation } from '../actions';

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  position: absolute;
  top: ${props => props.theme.structureSpacing};
  left: ${props => props.theme.structureSpacing};
  bottom: ${props => props.theme.structureSpacing};
  right: ${props => props.theme.structureSpacing};
`;

export const Message = styled.p`
  margin-bottom: 1em;
  font-size: 1.2em;
`;

export const ActionsWrapper = styled.div``;

export const CancelButton = styled(Button)`
  margin-right: 0.5em;
`;

export const ConfirmButton = styled(Button)``;

export const ConfirmationScreen = ({
  message,
  cancelConfirmationAction,
  confirmConfirmationAction,
}) => (
  <Wrapper id="confirmation-screen">
    <Message>{message}</Message>
    <ActionsWrapper>
      <CancelButton id="btn-cancel" onClick={cancelConfirmationAction}>
        Cancel
      </CancelButton>
      <ConfirmButton id="btn-confirm" onClick={confirmConfirmationAction}>
        Confirm
      </ConfirmButton>
    </ActionsWrapper>
  </Wrapper>
);

ConfirmationScreen.propTypes = {
  message: PropTypes.string.isRequired,
  cancelConfirmationAction: PropTypes.func.isRequired,
  confirmConfirmationAction: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  message: state.levelManager.confirmationMessage,
});

const mapDispatchToProps = dispatch => ({
  cancelConfirmationAction: bindActionCreators(cancelConfirmation, dispatch),
  confirmConfirmationAction: bindActionCreators(confirmConfirmation, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ConfirmationScreen);
