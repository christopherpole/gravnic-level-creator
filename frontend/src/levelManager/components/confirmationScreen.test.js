import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { spy } from 'sinon';

import { ConfirmationScreen, CancelButton, ConfirmButton } from './confirmationScreen';

configure({ adapter: new Adapter() });

describe('<ConfirmationScreen />', () => {
  let props;

  beforeAll(() => {
    props = {
      message: 'Are you sure?',
      cancelConfirmationAction: () => {},
      confirmConfirmationAction: () => {},
    };
  });

  it('Renders without exploding', () => {
    const confirmationScreen = shallow(<ConfirmationScreen {...props} />);

    expect(confirmationScreen).toHaveLength(1);
  });

  it('Matches the current snapshot', () => {
    const confirmationScreen = shallow(<ConfirmationScreen {...props} />);

    expect(toJson(confirmationScreen)).toMatchSnapshot();
  });

  it('Fires the cancel confirmation action when clicking on the "cancel" button', () => {
    const cancelConfirmationActionSpy = spy();
    const confirmationScreen = shallow(
      <ConfirmationScreen {...props} cancelConfirmationAction={cancelConfirmationActionSpy} />,
    );
    const btnCancel = confirmationScreen.find(CancelButton);
    btnCancel.simulate('click');

    expect(cancelConfirmationActionSpy.calledOnce).toBe(true);
  });

  it('Fires the cancel confirmation action when clicking on the "confirm" button', () => {
    const confirmConfirmationActionSpy = spy();
    const confirmationScreen = shallow(
      <ConfirmationScreen {...props} confirmConfirmationAction={confirmConfirmationActionSpy} />,
    );
    const btnConfirm = confirmationScreen.find(ConfirmButton);
    btnConfirm.simulate('click');

    expect(confirmConfirmationActionSpy.calledOnce).toBe(true);
  });
});
