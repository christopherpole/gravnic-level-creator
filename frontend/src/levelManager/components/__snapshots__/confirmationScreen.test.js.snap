// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationScreen /> Matches the current snapshot 1`] = `
<styled.div
  id="confirmation-screen"
>
  <styled.p>
    Are you sure?
  </styled.p>
  <styled.div>
    <Styled(Button)
      id="btn-cancel"
      onClick={[Function]}
    >
      Cancel
    </Styled(Button)>
    <Styled(Button)
      id="btn-confirm"
      onClick={[Function]}
    >
      Confirm
    </Styled(Button)>
  </styled.div>
</styled.div>
`;
